spec/controllers/admin/backups_controller_spec.rb

```diff
@@ -90,25 +90,43 @@
     describe ".show" do
 
       it "uses send_file to transmit the backup" do
+        token = EmailBackupToken.set
         path = File.join(Backup.base_directory, backup_filename)
         File.open(path, "w") { |f| f.write("hello") }
 
         Backup.create_from_filename(backup_filename)
 
         StaffActionLogger.any_instance.expects(:log_backup_download).once
 
-        get :show, id: backup_filename
+        get :show, id: backup_filename, token: token
 
         File.delete(path) rescue nil
+        EmailBackupToken.del
 
         expect(response.headers['Content-Length']).to eq("5")
         expect(response.headers['Content-Disposition']).to match(/attachment; filename/)
       end
 
+      it "return 404 when token is bad" do
+        path = File.join(Backup.base_directory, backup_filename)
+        File.open(path, "w") { |f| f.write("hello") }
+
+        Backup.create_from_filename(backup_filename)
+
+        get :show, id: backup_filename, token: "bad_value"
+
+        File.delete(path) rescue nil
```

Comment:
This needs to be run in an `ensure` block and not be rescued. Otherwise, it may silently fail and we end up not cleaning up after the test